src/QirRuntime/test/QIR-static/qir-test-ouput.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include <iostream>
+#include <sstream>
+
+#include "catch.hpp"
+
+#include "qirTypes.hpp"
+
+extern "C" void Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(QirString*);        // NOLINT
+
+
+// https://stackoverflow.com/a/5419388/6362941
+struct CoutRedirector
```

Comment:
Yeah, testability is another reason why implementing qis__message as std::cout might not be the right way to go. For example, if the restore of the buffer redirect fails, catch will continue running other tests but what will happen to catch's output to cout?

Reply:
Agreed.

I have written an email to Cesar (copy to you). I'll wait for what he says. Any recommendations about a better outputting and testing approach are welcome. 
For now I'm thinking about outputting not to C++'s `std::cout` and not to C's `stdout`, but to introduce an abstraction in front of those in the form of a logger (that buffers the output and then flushes to `std::cout` or `stdout`), such that for the test purposes I substitute the logger until the end of the scope. And the test will analyze the logger's buffer after the `Message()` function writes to it.

Reply:
I have updated my branch and addressed all the concerns except the last Chris' email about `Extensions.cs#L96`. I'll take a look at the email in more details tomorrow.

Reply:
To unblock Guen, I decided to postpone the changes (if any) inspired by Chris' email. 
I filed #518 to keep track of that. Marking this discussion as resolved in this PR.